health-checker/src/main/java/io/stargate/health/DataStoreHealthChecker.java

```diff
@@ -50,65 +49,55 @@
               ImmutableColumn.create(PK_COLUMN_NAME, Column.Kind.PartitionKey, Column.Type.Varchar),
               ImmutableColumn.create(VALUE_COLUMN_NAME, Column.Kind.Regular, Column.Type.Varchar))
           .build();
+  private final DataStoreFactory dataStoreFactory;
 
-  public DataStoreHealthChecker(BundleContext context) {
-    this.context = context;
+  public DataStoreHealthChecker(DataStoreFactory dataStoreFactory)
+      throws ExecutionException, InterruptedException {
+
+    this.dataStoreFactory = dataStoreFactory;
+    ensureTableExists(dataStoreFactory.createInternal());
   }
 
   @Override
   protected Result check() throws Exception {
     try {
-      ServiceReference<DataStoreFactory> dataStoreFactoryReference =
-          context.getServiceReference(DataStoreFactory.class);
-      if (dataStoreFactoryReference != null) {
-        DataStoreFactory dataStoreFactory = context.getService(dataStoreFactoryReference);
-        try {
-          DataStore dataStore = dataStoreFactory.createInternal();
-          ensureTableExists(dataStore);
-
-          // insert record
-          dataStore
-              .execute(
-                  dataStore
-                      .queryBuilder()
-                      .insertInto(KEYSPACE_NAME, TABLE_NAME)
-                      .value(PK_COLUMN_NAME, PK_COLUMN_VALUE)
-                      .value(VALUE_COLUMN_NAME, VALUE_COLUMN_VALUE)
-                      .build()
-                      .bind())
-              .get();
+      DataStore dataStore = dataStoreFactory.createInternal();
 
-          Future<ResultSet> rs =
+      // insert record
+      dataStore
+          .execute(
               dataStore
                   .queryBuilder()
-                  .select()
-                  .column(PK_COLUMN_NAME)
-                  .column(VALUE_COLUMN_NAME)
-                  .from(KEYSPACE_NAME, TABLE_NAME)
+                  .insertInto(KEYSPACE_NAME, TABLE_NAME)
+                  .value(PK_COLUMN_NAME, PK_COLUMN_VALUE)
+                  .value(VALUE_COLUMN_NAME, VALUE_COLUMN_VALUE)
```

Comment:
This value we're inserting should probably change from check to check by adding a UUID or timestamp. The pk value should probably be unique to a stargate instance too otherwise we could have N > 1 instances running all updating and reading a value for the same pk.

Reply:
I don't see a clear way of getting the id of the stargate instance. Do we generate a Stargate instance unique id?
Otherwise, we might try to use one of the addresses as pk, i.e.:
```
stargate.listen_address
```
but this setting is optional:
```
String listenAddress =
        System.getProperty("stargate.listen_address", InetAddress.getLocalHost().getHostAddress());
```
Otherwise, we could make a PK `UUID` and insert records with some low TTL. It seems that this solves both problems.
wdyt? 
implemented in:
https://github.com/stargate/stargate/pull/742/commits/a27e362d22b173473d155406cb2b73f92915f9df

Reply:
We could use the host ID from `system.local`, I guess

Reply:
Actually, if we access `system.local` for the host ID, we could combine the checks for `DataStore` availability (step 1) and storage availability (step 2) in the same checker, I guess. Step 2 could be optional (configurable). WDYT?

Reply:
I think that having two separate health checks is cleaner. Also, the `StorageHelathCheck` cab be disabled, whereas `DataStore` doesn't. So for me, both of those offer different functionality. The solution with UUID as PK solves those problems.

Reply:
Hmmm... I do not see how `StorageHelathCheck` can be disabled... I only see that DDL can be disabled, but the checker always runs... Did I miss something? :thinking:

Reply:
I will add a `"stargate.health_check.data_store.enabled"` for disabling that. 
I've proposed a solution here:
https://github.com/stargate/stargate/pull/742#discussion_r592987436
still waiting for feedback at which level do the enable/disable switch.

Reply:
_(added a comment in the other linked thread)_